File: Java/src/Solution001_100/Solution1.java
package Solution001_100;

/**
 * @author windows
 */
public class Solution1 {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            int temp = target - nums[i];
            for (int j = i + 1; j < nums.length; j++) {
                if(temp == nums[j]){
                    return new int[]{i, j};
                }
            }
        }
        return new int[]{0,0};
    }
}

File: C++/src/Solution1_100/Solution1.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution1.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> Solution1::twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> hash_table;
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        if (hash_table.count(complement)) {
            return {hash_table[complement], i};
        }
        hash_table[nums[i]] = i;
    }
    return {};
}
